### ISSUE TITLE
python3 support

### ISSUE THREAD

**Issue by gyorb** [Ericsson/codechecker#170]
test and update codechecker if needed
- the default interpreter for Arch linux is already python3
- next Ubuntu release (16.04) will also change the default interpreter to python3 https://wiki.ubuntu.com/Python/FoundationsXPythonVersions

Upgrade steps and requirements:
- [x] use unicode everywhere
- [x] [update logging style](https://docs.python.org/3/howto/logging-cookbook.html#use-of-alternative-formatting-styles)
- [ ] Minimal python3 version ([3.6.5-3 Ubuntu LTS 18.04](https://packages.ubuntu.com/search?suite=bionic&section=all&arch=any&keywords=python3&searchon=names))
- [ ] thrift compiler 0.11

**Comment by Xazax-hun** [comment]
thrift 0.10 will be released soon, and it is the first version to support python 3.

**Comment by Xazax-hun** [comment]
Thrift 0.10 is out and available in the pypi for python 3. Source: https://pypi.python.org/pypi/thrift/0.10.0

The last blocker to support python 3 should be gone.

**Comment by whisperity** [comment]
It's not gone.

Ubuntu 17.04 _Zesty_ and 17.10 _Artful_ still packages `thrift-0.9.x` and `python2` as `python` in their system.
Thrift _could_ be compiled from source, and even the generated files of `0.9.1` work with `0.10.0` installed from PIP work in Python2 mode.

However, updating Python and Thrift would help a lot in a pressing API issue about encoding large UTF-8 codepoints, such as #726.

**Comment by Xazax-hun** [comment]
Thrift 0.11 is already released but it looks like the community is not very quick with packaging it.
There is a 0.10 in debian experimental repositories: https://packages.debian.org/experimental/thrift-compiler
And it is not available in any of the ubuntu versions.

**Comment by whisperity** [comment]
Debian and Ubuntu isn't about getting the newest versions of everything, philosophically.

**Comment by LebedevRI** [comment]
Reminder that python2 will be fully and completely EOL in 1 month 12 days.

**Comment by gyorb** [comment]
yes, we know about it, thanks for the reminder

**Comment by hellow554** [comment]
Followup reminder: 

```
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
```

Is there a tracking issue on what's missing for python3?

**Comment by gyorb** [comment]
Hi,

There is no tracking issue, just a [project](https://github.com/Ericsson/codechecker/projects/8) with some notes I've collected.
I'm working on the port to [py3](https://github.com/gyorb/codechecker/tree/py3) it is not finished yet (The tests run on my machine successfully).
I still need to cleanup the commits and we are discussing in the team how to handle the generated thrift stubs (py3 stubs need at least thrift 0.11.0, but on Ubuntu LTS  that version is not available only 0.9.1).
Probably they will be published to pypi and npm but we still try to figure out what would be the easiest user/developer workflow in case of an api change.
